src/Compilers/CSharp/Portable/Binder/WithUsingNamespacesAndTypesBinder.cs

```diff
@@ -0,0 +1,358 @@
+                methods.RemoveDuplicates();
+            }
+
+        }
+
+        internal override void LookupSymbolsInSingleBinder(
+            LookupResult result, string name, int arity, ConsList<TypeSymbol>? basesBeingResolved, LookupOptions options, Binder originalBinder, bool diagnose, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        {
+            Debug.Assert(result.IsClear);
+
+            bool callerIsSemanticModel = originalBinder.IsSemanticModelBinder;
+
+            foreach (var typeOrNamespace in this.GetUsings(basesBeingResolved))
+            {
+                ImmutableArray<Symbol> candidates = Binder.GetCandidateMembers(typeOrNamespace.NamespaceOrType, name, options, originalBinder: originalBinder);
+                foreach (Symbol symbol in candidates)
+                {
+                    if (!IsValidLookupCandidateInUsings(symbol))
+                    {
+                        continue;
+                    }
+
+                    // Found a match in our list of normal using directives.  Mark the directive
+                    // as being seen so that it won't be reported to the user as something that
+                    // can be removed.
+                    var res = originalBinder.CheckViability(symbol, arity, options, null, diagnose, ref useSiteInfo, basesBeingResolved);
+                    if (res.Kind == LookupResultKind.Viable)
+                    {
+                        MarkImportDirective(typeOrNamespace.UsingDirectiveReference, callerIsSemanticModel);
+                    }
+
+                    result.MergeEqual(res);
+                }
+            }
+        }
+
+        private static bool IsValidLookupCandidateInUsings(Symbol symbol)
+        {
+            switch (symbol.Kind)
+            {
+                // lookup via "using namespace" ignores namespaces inside
+                case SymbolKind.Namespace:
+                    return false;
+
+                // lookup via "using static" ignores extension methods and non-static methods
+                case SymbolKind.Method:
+                    if (!symbol.IsStatic || ((MethodSymbol)symbol).IsExtensionMethod)
+                    {
+                        return false;
+                    }
+
+                    break;
+
+                // types are considered static members for purposes of "using static" feature
+                // regardless of whether they are declared with "static" modifier or not
+                case SymbolKind.NamedType:
+                    break;
+
+                // lookup via "using static" ignores non-static members
+                default:
+                    if (!symbol.IsStatic)
+                    {
+                        return false;
+                    }
+
+                    break;
+            }
+
+            return true;
+        }
+
+        protected override void AddLookupSymbolsInfoInSingleBinder(LookupSymbolsInfo result, LookupOptions options, Binder originalBinder)
```

Comment:
>AddLookupSymbolsInfoInSingleBinder [](start = 32, length = 34)

This is a combination of Imports.AddLookupSymbolsInfo and Imports.AddLookupSymbolsInfoInUsings